### PR TITLE
gcc@5: remove symlinks to lib

### PR DESCRIPTION
**Pull request by sjackman**
Merge PR https://github.com/Homebrew/brew/pull/13770 before merging this PR. Merge this PR before merging PR https://github.com/Homebrew/brew/pull/13733.
- https://github.com/Homebrew/brew/pull/13770
- https://github.com/Homebrew/brew/pull/13733

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- <https://github.com/Homebrew/brew/issues/13619>

### CODE REVIEW

File: Formula/gcc@11.rb [hunk 1]
```diff
@@ -121,6 +122,23 @@ def install
     # Work around GCC install bug
     # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105664
     rm_rf bin.glob("*-gcc-tmp")
+
+    # Add symlinks for libgcc, libgomp, libquadmath and libstdc++ so that bottles
+    # built in CI can find these libraries when using brewed gcc@5
```
Comment (SMillerDev) [inline Formula/gcc@11.rb:127]:
```suggestion
    # built in CI can find these libraries when using brewed gcc@11
```

**Review by danielnachun — DISMISSED** [review]

**Comment by carlocab** [comment]
Error inside an `on_linux` block on ARM macOS:
```
  Error: gcc@5: uninitialized constant OS::Linux
```

File: Formula/gcc@11.rb [hunk 2]
```diff
@@ -121,6 +122,23 @@ def install
...
+      lib.install_symlink lib/"gcc/#{version_suffix}/libgomp.so"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libgomp.a"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libgomp.so.1"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libquadmath.so"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libquadmath.a"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libquadmath.so.0"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libstdc++.so"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libstdc++.a"
+      lib.install_symlink lib/"gcc/#{version_suffix}/libstdc++.so.6"
+    end
```
Comment (carlocab) [inline Formula/gcc@11.rb:141]: Creating these symlinks and making them point to the right place seems to be better handled in `brew` than having to rev bump formulae whenever we need to change the preferred GCC version.

Comment (carlocab) [reply Formula/gcc@11.rb:141]: But if we must handle this in a formula, you probably need `link_overwrite`.

Comment (MikeMcQuaid) [reply Formula/gcc@11.rb:141]: Agreed that this feels like a Homebrew/brew thing.

**Review by MikeMcQuaid — COMMENTED** [review]
Agreed with @carlocab comments.

File: Formula/gcc@11.rb (hunk 2 above)
Comment (sjackman) [reply Formula/gcc@11.rb:141]: I agree. I don't have time myself to look at it until next week though. Feel free to tackle it if anyone has the inclination.

Comment (danielnachun) [reply Formula/gcc@11.rb:141]: We have previously discussed moving this to `brew` so it makes sense to do this now.  We already have `symlink_ld_so` in `extend/os/linux/install.rb` that is called as part of `perform_preinstall_checks`.  I think we can add `symlink_gcc_libs` to this which will make sure the `gcc` runtime libraries are linked whenever installing a formula.

**Comment by danielnachun** [comment]
I've opened https://github.com/Homebrew/brew/pull/13770 to automatically add the symlinks for whichever version of `gcc` is used in CI to `HOMEBREW_PREFIX/lib`.  We still a one-off revision bump of `gcc@5` to remove these symlinks, so I'm going to update this PR to reflect that change.

File: Formula/gcc@11.rb (hunk 2 above)
Comment (sjackman) [reply Formula/gcc@11.rb]: @danielnachun opened PR https://github.com/Homebrew/brew/pull/13770
- https://github.com/Homebrew/brew/pull/13770

**Review by MikeMcQuaid — APPROVED** [review]
LGTM whenever we're ready to ship this.

**Pull request closed without merging**